source/work-with-indexes.txt

```diff
@@ -0,0 +1,157 @@
+.. _cpp-work-with-indexes:
+
+=================
+Work with Indexes
+=================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+ 
+.. meta::
+   :keywords: query, optimization, efficiency
+
+Overview
+--------
+
+In this guide, you can learn how to use **indexes** with the {+driver-short+}.
+Indexes can improve the efficiency of queries and add additional functionality
+to querying and storing documents.
+
+Without indexes, MongoDB must scan every document in a collection to find the
+documents that match each query. These collection scans are
+slow and can negatively affect the performance of your application. However, if an
+appropriate index exists for a query, MongoDB can use the index to limit the
+documents it must inspect.
+
+Operational Considerations
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To improve query performance, build indexes on fields that appear often in
+your application's queries and operations that return sorted results. Each
+index that you add consumes disk space and memory when active, so we recommend
+that you track index memory and disk usage for capacity planning. In addition,
+when a write operation updates an indexed field, MongoDB updates the related
+index.
+
+Because MongoDB supports dynamic schemas, applications can query against fields
+whose names are not known in advance or are arbitrary. MongoDB 4.2 introduced
+:manual:`wildcard indexes </core/index-wildcard/>` to help support these
+queries. Wildcard indexes are not designed to replace workload-based index
+planning.
+
+For more information about designing your data model and choosing indexes appropriate for your application, see the
+:manual:`Data Modeling and Indexes </core/data-model-operations/#indexes>` guide
+in the {+mdb-server+} manual.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_mflix.movies`` collection
+from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:ref:`<cpp-get-started>`.
+
+Create an Index
+---------------
+
+MongoDB supports several different index types to support querying your data.
+The following sections describe the most common index types and provide sample
+code for creating each index type.
+
+Single-Field Indexes
+~~~~~~~~~~~~~~~~~~~~
+
+:manual:`Single-field indexes </core/index-single/>` are indexes with a
+reference to a single field within a collection's
+documents. They improve single field query and sort performance, and support
+:manual:`TTL Indexes </core/index-ttl>` that
```

Comment:
Q: Is this describing just one kind of TTL index, or is it a definition? If the latter, I'd do this:

```suggestion
:manual:`TTL Indexes </core/index-ttl>`, which
```

Reply:
I think it's the latter